### DISCUSSION TITLE
How to add internal-links in a note?

### DISCUSSION THREAD

**Discussion by YourNewuser** [TriliumNext/Trilium#3384] [Q&A]
Hello folks. 

In Trilium, I write my notes in the form of Questions and Answers where a Heading 6 is a question followed by an answer in Normal text. Example:

###### How to do X?
Some Normal text as answer

###### How to do Y with X?
Some normal text as answer

Suppose that in the second answer I want to refer myself to go read the question "How to do X" as a prerequisite. 
How would I go about doing this? Ideally, I'm looking for an internal-only link.

In an HTML file, one can refer to an internal HTML link by referencing the `id` attribute of the heading and adding anchor link wherever needed with `href='#id'`. In an Office app such as LibreOffice I can create a bookmark and than add an internal link which would reference that bookmark.

In Trilium I don't know how to do something like that. I think this is doable because Table of Contents knows how to scroll to the referenced heading. That logic perhaps can be ported as a new internal-link-only functionality?

Please do let me know if you have a way to add internal-only links.

**Comment by jaroet** [comment]
You could just add text notes below the main note with one question / answer per note. When you then go to the main note the child notes are shown as cards. When you add some extra info as attributes you could also have saved searches for different topics.
Go to the Inbox note in the "Trilium Demo" content that is included when you install it. 

You could also give the main note the type of book and then all child notes are shown in order in full width. See the "Formatting examples" in the Trilium demo content.

**Reply by YourNewuser** [reply]
Thank you. I think this approach is not something I can use currently. Reasons:

1. I'd be losing Table of Contents (TOC). TOC is important to me.
2. Second, I'll have to split lots of my Q&As into child notes, and that's too much work. Not to mention for each Q&A I'd have to create a new note. And I create many Q&As for learning purposes. :)

I think if the number of Q&As are manageable than your approach would be a good workaround.

**Comment by zadam** [comment]
Hi, such a feature request already exists - see e.g. https://github.com/zadam/trilium/issues/1021

**Reply by YourNewuser** [reply]
Cool. Somebody has a thread for it. 

I'm sorry I didn't search this in the issue earlier. I only limited my search to "discussion", else I would't have asked here. :)
